### ISSUE TITLE
Project search (M-m /) fails to open file on second search attempt

### ISSUE THREAD

**Issue by Veebers** [syl20bnr/spacemacs#10465]
#### Description :octocat:
Opening a file found while searching (M-m /) for something fails on the second time as it prepends a directory name from the first search success.

For instance; if you open a file (from search results) a/foo.txt and then try open a file (again, from search results) b/foo.txt (that contains the text you've searched for) it will attempt to open "a/b/foo.txt" not "b/foo.txt" and thus opens a new file.

#### Reproduction guide :beetle:
- I've encountered this while searching within a project, so:
  - mkdir -p /tmp/testing/{a,b}
  - touch /tmp/testing/{a,b}/foo.txt
  - echo "Hello a" > /tmp/testing/a/foo.txt
  - echo "Hello b" > /tmp/testing/b/foo.txt
  - cd /tmp/testing && git init . && git add . && git commit -am "Initial commit"
  - emacs /tmp/testing &

  - First search:
    - M-m /
    - type: Hello
    - Hit enter to open the first result (a/foo.txt)
    - Observe that this works and opens the correct file.

  - Second search:
    - M-m /
    - type: Hello
    - Move down to the second option (C-n) b/foo.txt and hit enter
    - Observe this  *should* open b/foo.txt but attempts to open a/b/foo.txt and thus opens a new file

*Observed behaviour:* :eyes: :broken_heart:
On the 2nd search it attempts to open a file directory that has the original search result directory pre-pended to it. (i.e. a/ is added to the result to give the incorrect a/b/foo.txt)

*Expected behaviour:* :heart: :smile:
It should open the correct path (b/foo.txt) with no extra path parts added to it at all.

#### System Info :computer:
- OS: gnu/linux
- Emacs: 25.2.2
- Spacemacs: 0.200.13
- Spacemacs branch: master (rev. c7a103a7)
- Graphic display: t
- Distribution: spacemacs
- Editing style: emacs
- Completion: ivy
- Layers:
```elisp
(markdown python go ivy auto-completion better-defaults emacs-lisp git version-control themes-megapack)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GSETTINGS NOTIFY ACL LIBSELINUX GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB TOOLKIT_SCROLL_BARS GTK3 X11

**Comment by duianto** [comment]
This seems to be a duplicate of these two issues:
spacemacs/search-project-ag expends with wrong file path when open selected candidate #9552
[question] SilverSearcher open empty buffer. #9652

The last comment on both pages, says that this issue has been fixed on the Spacemacs `develop` branch.

When I try to replicate the issue, on both the `master` and `develop` branches, then neither of the `foo.txt` files open from the `M-m /` prompt.

- With a dired buffer open, at the `testing` directory
- Pressing `M-m /`, opens a prompt:
```
2    ag from [c:/temp/testing/]:
```
- Searching for `hello`, shows two candidates:
```
Hello a^M
Hello b^M
```

Pressing `RET` on either of them, just closes the prompt, without opening the `foo.txt` file.

I have updated the packages, so I'm using the latest version of ivy, and there are no ivy specific configurations, I just commented out `helm` and uncommented `ivy` in the
`dotspacemacs-configuration-layers` variable.

Both `foo.txt` files can be opened without issue, from the dired buffer, or with `M-m f f`.

#### System Info :computer:
- OS: windows-nt
- Emacs: 25.3.1
- Spacemacs: 0.200.13
- Spacemacs branch: master (rev. c7a103a7)
- Spacemacs: 0.300.0
- Spacemacs branch: develop (rev. 0fa3658c)
- Graphic display: t
- Distribution: spacemacs
- Editing style: emacs
- Completion: ivy
- Layers:
```elisp
(autohotkey auto-completion emacs-lisp git ivy org spell-checking treemacs version-control)
```
- System configuration features: XPM JPEG TIFF GIF PNG RSVG SOUND NOTIFY ACL GNUTLS LIBXML2 ZLIB TOOLKIT_SCROLL_BARS MODULES

**Comment by Veebers** [comment]
I tried using develop and the issue goes away (I've now changed to using develop in my main .emacs.d/ and not just as a test.)

Unfortunately the issue persists with commands that are not the default. In this instance it's `counsel-projectile-find-file`. i.e.:

  - M-m p f  -> Type file name and hit enter, it works!
  - M-m p f -> Type file name -> M-o j (other window) I see the same behaviour (i.e. an extra directory prepended to the path of the file and thus opens an empty file.)
